tests/datastreams/test_datastreams.py

```diff
@@ -144,3 +146,154 @@ def test_piping_readers(app, zip_file, json_element):
+                                <subfield code="b">s</subfield>
+                                <subfield code="2">gndgen</subfield>
+                            </datafield>
+                            <datafield tag="075" ind1=" " ind2=" ">
+                                <subfield code="b">saz</subfield>
+                                <subfield code="2">gndspec</subfield>
+                            </datafield>
+                            <datafield tag="079" ind1=" " ind2=" ">
+                                <subfield code="a">g</subfield>
+                                <subfield code="q">s</subfield>
+                            </datafield>
+                            <datafield tag="150" ind1=" " ind2=" ">
+                                <subfield code="a">Rundbogenhalle</subfield>
+                            </datafield>
+                            <datafield tag="450" ind1=" " ind2=" ">
+                                <subfield code="a">Bogenhalle</subfield>
+                            </datafield>
+                            <datafield tag="550" ind1=" " ind2=" ">
+                                <subfield code="0">(DE-101)040230236</subfield>
+                                <subfield code="0">(DE-588)4023023-5</subfield>
+                                <subfield code="0">https://d-nb.info/gnd/4023023-5</subfield>
+                                <subfield code="a">Halle</subfield>
+                                <subfield code="4">obge</subfield>
+                                <subfield code="4">https://d-nb.info/standards/elementset/gnd#broaderTermGeneric</subfield>
+                                <subfield code="w">r</subfield>
+                                <subfield code="i">Oberbegriff generisch</subfield>
+                            </datafield>
+                            <datafield tag="670" ind1=" " ind2=" ">
+                                <subfield code="a">Stahlbetonbauwerke mit großer Spannweite, eingesetzt im Industriebau, z.b.Paketposthalle München; teilweise heute unter Denkmalschutz</subfield>
+                            </datafield>
+                        </record>
+                    </metadata>
+                </record>
+            </ ListRecords>
+        </OAI-PMH>
+    """
+    httpserver.expect_request("/oai/repository").respond_with_data(
+        response_data=response_data, mimetype="application/xml"
+    )
+    ds_config = {
+        "readers": [
+            {
+                "type": "oai-pmh",
+                "args": {
+                    "base_url": httpserver.url_for("/oai/repository"),
+                    "metadata_prefix": "MARC21plus-1-xml",
+                    "set": "authorities:sachbegriff",
+                    "from_date": "2024-01-01T09:00:00Z",
+                    "until_date": "2024-01-31T10:00:00Z",
+                },
+            },
+        ],
+        "writers": [{"type": "test"}],
+    }
```

Comment:
Since you are testing `OAIPMHReader` specifically, I would say that defining a full config/factory here is not necessary.
You could:
1. Move these tests from `tests/datastreams/test_datastreams.py` to `tests/datastreams/test_readers.py`
2. Change the test to look more like other reader tests (calling directly the constructor and then the `read` method).
3. Remove the definition of the OAIPMHReader in `tests/datastreams/conftest.py`

Reply:
Done.